test/gateway/eip20_cogateway/progress_mint.js

```diff
@@ -0,0 +1,398 @@
+        await helper.setCoGateway(testEIP20CoGateway.address);
+        
+        messageHash = await testEIP20CoGateway.setStakeMessage.call(
+            intentHash,
+            nonce,
+            gasPrice,
+            gasLimit,
+            hashLock,
+            staker,
+        );
+        await testEIP20CoGateway.setStakeMessage(
+            intentHash,
+            nonce,
+            gasPrice,
+            gasLimit,
+            hashLock,
+            staker,
+        );
+    
+        await testEIP20CoGateway.setMints(messageHash, beneficiary, amount);
+        
+    });
+    
+    it('should mint for non-zero facilitator reward', async function () {
+        
+        await testEIP20CoGateway.setInboxStatus(
+            messageHash,
+            MessageStatusEnum.Declared,
+        );
+        
+        let progressMintValues = await testEIP20CoGateway.progressMint.call(
+            messageHash,
+            unlockSecret,
+            {from: facilitator},
+        );
+        
+        let expectedMintedToken = new BN(100),
+            expectedReward = new BN(100);
+        
+        assert.strictEqual(
+            progressMintValues.beneficiary_,
+            beneficiary,
+            `Beneficiary address should be ${beneficiary}`,
+        );
+        
+        assert.strictEqual(
+            amount.eq(progressMintValues.stakeAmount_),
+            true,
+            `Staked amount should be ${amount}.`,
+        );
+        
+        assert.strictEqual(
+            expectedMintedToken.eq(progressMintValues.mintedAmount_),
+            true,
+            `Minted amount should be ${expectedMintedToken}.`,
+        );
+        
+        assert.strictEqual(
+            expectedReward.eq(progressMintValues.rewardAmount_),
+            true,
+            `Reward to facilitator should be ${expectedReward}.`,
+        );
+        
+        let response = await testEIP20CoGateway.progressMint(
+            messageHash,
+            unlockSecret,
+            {from: facilitator},
+        );
+        
+        let facilitatorBalance = await testUtilityToken.balanceOf(facilitator);
+        let beneficiaryBalance = await testUtilityToken.balanceOf(beneficiary);
+        
+        assert.strictEqual(
+            facilitatorBalance.eq(expectedReward),
+            true,
+            `Facilitator reward should be ${expectedReward}.`,
+        );
+        
+        assert.strictEqual(
+            beneficiaryBalance.eq(amount.sub(expectedReward)),
+            true,
+            `Beneficiary balance should be ${amount.sub(expectedReward)}.`
+        );
+        
+        let expectedEvent = {
+            MintProgressed: {
+                _messageHash: messageHash,
+                _staker: staker,
+                _stakeAmount: amount,
+                _mintedAmount: expectedMintedToken,
+                _rewardAmount: expectedReward,
+                _proofProgress: true,
```

Comment:
```suggestion
                _proofProgress: false,
```
This should return false in case of progress with hashlock.